google-cloud-storage/lib/google/cloud/storage/bucket/lifecycle.rb

```diff
@@ -243,6 +287,10 @@ def freeze
           #   Values include `STANDARD`, `NEARLINE`, `COLDLINE`, and `ARCHIVE`.
           #   `REGIONAL`, `MULTI_REGIONAL`, and `DURABLE_REDUCED_AVAILABILITY`
           #   are supported as legacy storage classes.
+          # @attr [String,DateTime] noncurrent_time_before A timestamp string in
```

Comment:
For cases like this when an attribute can be multiple types, can we clarify the behavior, e.g. when you read the field, when do you get a String vs a DateTime? (I assume that when you load data from the backend, you get a DateTime, but when you set a value locally, you get back whatever type you used to set it. But that has the potential of being confusing, because then you're kind of leaking when syncs happen.)

Reply:
Great suggestion, will do.